src/pad/src/RDLRouter.cpp

```diff
@@ -128,9 +129,141 @@ class RDLRouterGoalVisitor : public boost::default_astar_visitor
+                                  [this](odb::dbITerm* other) {
+                                    return iterm_->getInst()
+                                           == other->getInst();
+                                  }),
+                   terminals_.end());
+
+  const odb::Point iterm_center = iterm_->getBBox().center();
+  std::stable_sort(terminals_.begin(),
+                   terminals_.end(),
+                   [&iterm_center](odb::dbITerm* lhs, odb::dbITerm* rhs) {
+                     const bool lhs_cover = RDLRouter::isCoverTerm(lhs);
+                     const bool rhs_cover = RDLRouter::isCoverTerm(rhs);
+                     // sort non-cover terms first
+                     if (lhs_cover == rhs_cover) {
+                       return odb::Point::squaredDistance(
+                                  iterm_center, lhs->getBBox().center())
+                              < odb::Point::squaredDistance(
+                                  iterm_center, rhs->getBBox().center());
+                     }
+                     if (rhs_cover) {
+                       return true;
+                     }
+                     return false;
+                   });
+
+  resetRoute();
+}
+
+odb::dbITerm* RDLRoute::peakNextTerminal() const
+{
+  return *next_;
+}
+
+odb::dbITerm* RDLRoute::getNextTerminal()
+{
+  odb::dbITerm* iterm = peakNextTerminal();
+
+  next_++;
+
+  return iterm;
+}
+
+bool RDLRoute::compare(const std::shared_ptr<RDLRoute>& other) const
+{
+  const auto lhs_priority = -getPriority();
+  const auto rhs_priority = -other->getPriority();
+  if (!hasNextTerminal() || !other->hasNextTerminal()) {
+    return lhs_priority > rhs_priority;
+  }
+
+  const auto& lhs_shortest = peakNextTerminal();
+  const auto lhs_dist = odb::Point::squaredDistance(
+      getTerminal()->getBBox().center(), lhs_shortest->getBBox().center());
+
+  const auto& rhs_shortest = other->peakNextTerminal();
+  const auto rhs_dist
+      = odb::Point::squaredDistance(other->getTerminal()->getBBox().center(),
+                                    rhs_shortest->getBBox().center());
+
+  return std::tie(lhs_priority, lhs_dist) > std::tie(rhs_priority, rhs_dist);
+}
+
+void RDLRoute::setRoute(
+    const std::vector<grid_vertex>& vertex,
+    const std::set<std::tuple<odb::Point, odb::Point, float>>& removed_edges,
+    const RouteTarget* source,
+    const RouteTarget* target)
+{
+  route_vertex_ = vertex;
+  route_edges_ = removed_edges;
+  route_source_ = source;
+  route_dest_ = target;
+}
+
+void RDLRoute::resetRoute()
+{
+  route_vertex_.clear();
+  route_edges_.clear();
+  route_source_ = nullptr;
+  route_dest_ = nullptr;
+  route_pending_ = true;
+
+  next_ = terminals_.begin();
+}
+
+bool RDLRoute::isIntersecting(RDLRoute* other, int extent) const
+{
+  // check current next_
+  // if at end of next, select begin + priority offset (modulo)
+  const int dst_idx = priority_ % terminals_.size();
+  odb::dbITerm* dst = terminals_[dst_idx];
```

Comment:
I don't follow this.

Reply:
This is just a simple way to "perturb" the routing calculations. The idea is just to go to a new destination iterm, if possible. I'm sure there are many improvements possible to the "what to ripup"-code, but this avoids the ripup from being identically calculated on a route that fails more than once.

Reply:
Ok but please add a more descriptive comment.